### PR TITLE
Feature/users route

### PR DESCRIPTION
**Pull request by Naomi-Eckhoff**

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
We will need to change any references to `Users, Teachers, Students` to singular case; `User, Teacher, Student`. 1/03/2022: not sure this is necessary.

File: controllers/api/users-routes.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    req.session.destroy(() => {
+      res.status(204).end();
+    });
+  }
+  else {
+    res.status(404).end();
+  }
+});
+
+router.put('/:id', (req, res) => {
```
Comment (HighDynamics) [inline controllers/api/users-routes.js]: Let's add the withAuth middleware to this route

File: controllers/api/users-routes.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+      }
+      res.json(dbUsersData);
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json(err);
+    });
+});
+
+router.delete('/:id', (req, res) => {
```
Comment (HighDynamics) [inline controllers/api/users-routes.js]: Let's add the withAuth middleware to this route

**Review by HighDynamics — CHANGES_REQUESTED** [review]
Two spots to add the `withAuth` middleware

File: controllers/api/users-routes.js (hunk 1 above)
Comment (Naomi-Eckhoff) [reply controllers/api/users-routes.js]: added

File: controllers/api/users-routes.js (hunk 2 above)
Comment (Naomi-Eckhoff) [reply controllers/api/users-routes.js]: added

**Review by ghost — APPROVED** [review]

**Review by HighDynamics — APPROVED** [review]

**Pull request merged**